pkg/scheduler/flavorassigner/tas_flavorassigner.go

```diff
@@ -20,63 +20,69 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/go-logr/logr"
 	"k8s.io/utils/ptr"
 
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 	"sigs.k8s.io/kueue/pkg/cache"
+	"sigs.k8s.io/kueue/pkg/features"
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
-func assignTopology(log logr.Logger,
-	psAssignment *PodSetAssignment,
+// TopologyRequestsFor - returns the TopologyRequests of the workload
+func (a *Assignment) TopologyRequestsFor(wl *workload.Info,
+	requests []workload.PodSetResources,
+	cq *cache.ClusterQueueSnapshot) map[kueue.ResourceFlavorReference][]cache.TASPodSetRequests {
+	var tasRequests map[kueue.ResourceFlavorReference][]cache.TASPodSetRequests
+	if features.Enabled(features.TopologyAwareScheduling) {
+		for i, podSet := range wl.Obj.Spec.PodSets {
+			if podSet.TopologyRequest != nil {
+				psAssignment := a.podSetByName(podSet.Name)
+				if psAssignment.Status.IsError() {
+					// There is no resource quota assignment for the PodSet - no need to check TAS.
+					continue
+				}
+				psTASRequest, err := podSetTopologyRequest(psAssignment, wl, cq, i)
+				if err != nil {
+					psAssignment.error(err)
+				} else if psTASRequest != nil {
+					if tasRequests == nil {
+						tasRequests = make(map[kueue.ResourceFlavorReference][]cache.TASPodSetRequests)
+					}
```

Comment:
Can we avoid such an initialization

Reply:
Yes, I did it to avoid allocation for the map, but you can say it is premature optimization. Alternatively we could iterate over the PodSets and just check if any of them requires TAS. before the main loop.